components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -1208,10 +1208,13 @@ public String getGraphqlSchemaDefinition(APIIdentifier apiId) throws APIManageme
     public String getOpenAPIDefinition(Identifier apiId) throws APIManagementException {
         String apiTenantDomain = getTenantDomain(apiId);
         String swaggerDoc = null;
+        boolean tenantFlowStarted = false;
         try {
             Registry registryType;
             //Tenant store anonymous mode if current tenant and the required tenant is not matching
             if (this.tenantDomain == null || isTenantDomainNotMatching(apiTenantDomain)) {
+                startTenantFlow(apiTenantDomain);
```

Comment:
This will not work for cross tenants with logged in users.

Reply:
Yes, I also thinks still registry would be the wrong one, unless we explicitly load the registry

Reply:
We tested it. This works as expected.